src/main/java/com/javadiscord/jdi/internal/api/DiscordRequestBuilder.java

```diff
@@ -0,0 +1,84 @@
+package com.javadiscord.jdi.internal.api;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class DiscordRequestBuilder {
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private String method;
+    private String url;
+    private String body;
+    private DiscordResponseFuture future;
+    private final Map<String, Object> headers = new HashMap<>();
+
+    public DiscordRequestBuilder addHeader(String name, String value) {
```

Comment:
`setHeader` implies we might be setting the entire object (especially since there's the implied setter/getter pattern). This method is appending to the headers map. As such, I think addHeader is a better name. What do you think?

Reply:
> implies we might be setting the entire object

that is actually more the case for `setHeaders` since the variable is named  `headers` and the parameters makes it look like its setting the entire object. Ig we should rename both to `putHeader` and `putHeaders`

Reply:
Done